src/PrestaShopBundle/Form/Admin/Improve/Payment/Preferences/PaymentModuleCurrencyRestrictionsType.php

```diff
@@ -0,0 +1,135 @@
+ */
+class PaymentModuleCurrencyRestrictionsType extends PaymentModuleRestrictionsParentType
+{
+    public const CUSTOMER_CURRENCY = -1;
+    public const SHOP_DEFAULT_CURRENCY = -2;
+    /**
+     * @var array
+     */
+    private $currencyChoices;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param array $locales
+     * @param array $paymentModules
+     * @param array $currencyChoices
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        array $paymentModules,
+        array $currencyChoices
+    ) {
+        parent::__construct($translator, $locales, $paymentModules);
+
+        $this->currencyChoices = $currencyChoices;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('currency_restrictions', MaterialMultipleChoiceTableType::class, [
+                'label' => $this->trans('Currency restrictions', 'Admin.Payment.Feature'),
+                'table_label' => $this->trans('Currency restrictions', 'Admin.Payment.Feature'),
+                'help' => $this->trans(
+                    'Please select available payment modules for each currency.',
+                    'Admin.Payment.Help'
+                ),
+                'required' => false,
+                'choices' => $this->getCurrencyChoices(),
+                'multiple_choices' => $this->getCurrencyChoicesForPaymentModules(),
+                'headers_fixed' => true,
+            ]);
+    }
+
+    /**
+     * Get multiple currency choices for payment modules.
+     *
+     * @return array
+     */
+    private function getCurrencyChoicesForPaymentModules(): array
+    {
+        $choices = [];
+
+        foreach ($this->paymentModules as $paymentModule) {
+            $moduleInstance = $paymentModule->getInstance();
+
+            $allowMultipleCurrencies = true;
+            $currencyChoices = $this->currencyChoices;
+            if ('radio' === $moduleInstance->currencies_mode) {
+                $allowMultipleCurrencies = false;
+                $currencyChoices = $this->getCurrencyChoices();
+            }
+            $choices[] = [
+                'name' => $paymentModule->get('name'),
+                'label' => $paymentModule->get('displayName'),
+                'multiple' => $allowMultipleCurrencies,
+                'choices' => $currencyChoices,
+            ];
+        }
+
+        return $choices;
+    }
+
+    /**
+     * Get currency choices with specific additional choices.
+     *
+     * @return array
+     */
+    private function getCurrencyChoices(): array
+    {
+        return array_merge(
+            $this->currencyChoices,
+            $this->getAdditionalCurrencyChoices()
+        );
+    }
+
+    /**
+     * Get payment preferences specific currency choices.
+     *
+     * @return array<string, int>
+     */
+    private function getAdditionalCurrencyChoices(): array
```

Comment:
I checked it and I don't think it's a good idea because we do not always add those 2 currency choices.

When when` $moduleInstance->currencies_mode != 'radio'` those 2 options should not be added.